gtsfm/utils/ellipsoid.py

```diff
@@ -0,0 +1,179 @@
+    cutoff_mag = 1.65 * (std_mag) + mean_mag
+
+    indices_to_keep = np.where(np.linalg.norm(point_cloud, axis=1) < cutoff_mag)
+    points_filtered = point_cloud[indices_to_keep]
+    return points_filtered
+
+
+def get_rotation_matrix(point_cloud: np.ndarray) -> np.ndarray:
+    """Applies SVD on a point cloud. The resulting V contains the rotation matrix required to align the 3 major axes of
+    the point cloud's ellipsoid with the x, y, z coordinate axes.
+
+    Args:
+        point_cloud: point cloud of shape N x 3.
+
+    Returns:
+        The rotation matrix, shape 3 x 3, required to align points with the x, y, and z axes.
+
+    Raises:
+        TypeError: if point cloud is not of shape N x 3.
+    """
+    if point_cloud.shape[1] != 3:
+        raise TypeError("Point Cloud should be 3 dimensional")
+
+    U, S, Vt = np.linalg.svd(point_cloud, full_matrices=True)
+    wuprightRw = np.linalg.inv(Vt.T)
+    return wuprightRw
+
+
+def apply_ellipsoid_rotation(wuprightRw: np.ndarray, point_cloud: np.ndarray) -> np.ndarray:
+    """Applies a rotation on the centered, filtered point cloud.
+
+    Args:
+        wuprightRw: rotation matrix, shape 3 x 3.
+        point_cloud: point cloud, shape N x 3.
+
+    Returns:
+        A transformed, aligned point cloud of shape N x 3.
+
+    Raises:
+        TypeError: if rot matrix isn't 3x3 or centered_pc is not of shape N x 3.
+    """
+    if wuprightRw.shape[0] != 3 or wuprightRw.shape[1] != 3:
+        raise TypeError("Rotation Matrix should be of shape 3 x 3")
+    if point_cloud.shape[1] != 3:
+        raise TypeError("Point Cloud shoud be 3 dimensional")
+
+    # First apply the rotation matrix to the point cloud. Then swap x and z axes for visual alignment.
+    swap_x_z = np.array([[0, 0, 1], [0, 1, 0], [-1, 0, 0]])
+    rotated_points = swap_x_z @ wuprightRw @ point_cloud.T
+
+    return rotated_points.T
+
+
+def transform_camera_frustums(
+    wti: np.ndarray, wRi_quaternion: np.ndarray, means: np.ndarray, wuprightRw: np.ndarray
+) -> Tuple[List[float], List[float]]:
+    """Transforms the camera frustums in a similar manner as the point cloud.
+
+    Args:
+        wti: camera pose translation, list of length 3.
+        wRi_quaternion: camera pose rotation, list of length 4.
+        means: the mean x,y,z coordinates of point cloud, array of shape (3.).
+        wuprightRw: rotation matrix to align point cloud with x,y,z axes, shape 3 x 3.
+
+    Returns:
+        Two lists. One list, of length 4, representing the quaternion for camera frustum rotation. The second list,
+        of length 3, representing the camera frustum translation.
+    """
+
+    tx, ty, tz = wti
+    qw, qx, qy, qz = wRi_quaternion
+    gtsfm_rot = R.from_quat([qx, qy, qz, qw]).as_matrix()
+    gtsfm_tran = np.array([tx, ty, tz])
+
+    gtsfm_pose = np.eye(4)
+    gtsfm_pose[:3, :3] = gtsfm_rot
+    gtsfm_pose[:3, 3] = gtsfm_tran
```

Comment:
we shouldn't name it as pose -- we can make it clearer to the reader per the gtsam conventions
`wTi` means it's the i'th camera's pose in the world frame